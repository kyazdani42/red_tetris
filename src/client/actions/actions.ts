import { Action, ActionFunctionAny, createAction } from 'redux-actions';

import { AppState } from '../reducers/app';

import {
  CREATE_ROOM,
  HANDLE_KEY_PRESS,
  JOIN_ROOM,
  LEAVE_ROOM,
  SET_GAME_DATA,
  SET_KEY,
  SET_OPTIONS,
  SET_PLAYER_NAME,
  SET_ROOMS,
  SET_SOCKET,
} from '../constants';

export const setRooms: ActionFunctionAny<Action<RoomType[]>> = createAction(SET_ROOMS);

export const createRoom: ActionFunctionAny<Action<string>> = createAction(CREATE_ROOM);
export const joinRoom: ActionFunctionAny<Action<undefined>> = createAction(JOIN_ROOM);
export const leaveRoom: ActionFunctionAny<Action<undefined>> = createAction(LEAVE_ROOM);

export const setSocket: ActionFunctionAny<Action<SocketIOClient.Socket>> = createAction(SET_SOCKET);

export const setGameData: ActionFunctionAny<Action<GameProps>> = createAction(SET_GAME_DATA);
export const setKey: ActionFunctionAny<Action<keyType>> = createAction(SET_KEY);
export const handleKeyPress: ActionFunctionAny<Action<keyType>> = createAction(HANDLE_KEY_PRESS);

export const setPlayerName: ActionFunctionAny<Action<string>> = createAction(SET_PLAYER_NAME);
export const setOptions: ActionFunctionAny<Action<AppState['options']>> = createAction(SET_OPTIONS);
