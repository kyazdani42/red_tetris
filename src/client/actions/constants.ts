export const SET_ROOMS = 'SET_ROOMS';
export const JOIN_ROOM = 'HANDLE_JOIN_ROOM';

export const CREATE_ROOM = 'CREATE_ROOM';
export const LEAVE_ROOM = 'LEAVE_ROOM';

export const SET_SOCKET = 'SET_SOCKET';

export const SET_GAME_DATA = 'SET_GAME_DATA';
