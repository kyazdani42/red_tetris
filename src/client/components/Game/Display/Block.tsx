import * as React from 'react';

export const NextBlock = (color: string, key: string) => (
  <div
    style={{
      backgroundColor: color,
      border: '1px solid black',
      width: 'calc(25% - 2px)',
      height: 'calc(25% - 2px)'
    }}
    key={key}
  />
);

export const Block = (color: string, key: string) => (
  <div
    style={{
      backgroundColor: color,
      border: '1px solid black',
      width: 'calc(10% - 2px)',
      height: 'calc(5% - 2px)'
    }}
    key={key}
  />
);
