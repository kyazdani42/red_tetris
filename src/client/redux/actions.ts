import { Action, ActionFunctionAny, createAction } from 'redux-actions';

import { AppState } from './reducer';

import {
  CREATE_ROOM,
  GET_SCORES,
  HANDLE_KEY_PRESS,
  JOIN_ROOM,
  LEAVE_ROOM,
  SET_ERROR,
  SET_GAME_DATA,
  SET_KEY,
  SET_MODAL,
  SET_MUSIC_PLAYING,
  SET_OPTIONS,
  SET_PLAYER_NAME,
  SET_ROOMS,
  SET_SOCKET,
  SET_TOKEN,
} from './constants';

export const setRooms: ActionFunctionAny<Action<RoomType[]>> = createAction(SET_ROOMS);

export const createRoom: ActionFunctionAny<Action<string>> = createAction(CREATE_ROOM);
export const joinRoom: ActionFunctionAny<Action<undefined>> = createAction(JOIN_ROOM);
export const leaveRoom: ActionFunctionAny<Action<undefined>> = createAction(LEAVE_ROOM);
export const setModal: ActionFunctionAny<Action<boolean>> = createAction(SET_MODAL);

export const setSocket: ActionFunctionAny<Action<SocketIOClient.Socket>> = createAction(SET_SOCKET);

export const setGameData: ActionFunctionAny<Action<GameProps>> = createAction(SET_GAME_DATA);
export const setKey: ActionFunctionAny<Action<keyType>> = createAction(SET_KEY);
export const handleKeyPress: ActionFunctionAny<Action<keyType>> = createAction(HANDLE_KEY_PRESS);

export const setPlayerName: ActionFunctionAny<Action<string>> = createAction(SET_PLAYER_NAME);
export const setError: ActionFunctionAny<Action<string | null>> = createAction(SET_ERROR);
export const setOptions: ActionFunctionAny<Action<AppState['options']>> = createAction(SET_OPTIONS);
export const setToken: ActionFunctionAny<Action<string>> = createAction(SET_TOKEN);
export const setMusicPlaying: ActionFunctionAny<Action<undefined>> = createAction(SET_MUSIC_PLAYING);

export const getScores: ActionFunctionAny<Action<undefined>> = createAction(GET_SCORES);
