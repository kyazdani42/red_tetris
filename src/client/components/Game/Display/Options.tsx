import * as React from 'react';
import { connect } from 'react-redux';

import { setOptions } from '../../../actions/actions';
import { State } from '../../../store';
import { OptionLabelWrapper, OptionsWrapper } from './styles';

interface Props {
  options: Options;
  dispatchSetOptions: (options: Options) => void;
}

const handleDispatch = (params: Props) => (option: { name: string; on: boolean }) =>
  params.dispatchSetOptions({ ...params.options, [option.name]: option.on });

export const Options: React.SFC<Props> = ({ options, dispatchSetOptions }) => {
  const dispatch = handleDispatch({ options, dispatchSetOptions });
  return (
    <OptionsWrapper>
      {checkbox(options.invisible, 'invisible', dispatch)}
      {checkbox(options.mirror, 'mirror', dispatch)}
      {checkbox(options.reverse, 'reverse', dispatch)}
    </OptionsWrapper>
  );
};

const checkbox = (
  checked: boolean,
  name: string,
  dispatch: (option: { name: string; on: boolean }) => void
) => (
  <OptionLabelWrapper>
    <label>{name}</label>
    <input type="checkbox" checked={checked} onChange={() => dispatch({ name, on: !checked })} />
  </OptionLabelWrapper>
);

const mapStateToProps = (state: State) => ({
  options: state.app.options
});

const mapDispatchToProps = (dispatch: any) => ({
  dispatchSetOptions: (options: Options) => dispatch(setOptions(options))
});

export default connect(
  mapStateToProps,
  mapDispatchToProps
)(Options);
