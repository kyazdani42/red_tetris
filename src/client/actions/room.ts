import { Action, ActionFunctionAny, createAction } from 'redux-actions';
import { RoomType } from '../types';
import { HANDLE_JOIN_ROOM, SET_ROOMS, SET_SELECTED_ROOM } from './constants';

export const setRooms: ActionFunctionAny<Action<RoomType[]>> = createAction(SET_ROOMS);
export const setSelectedRoom: ActionFunctionAny<Action<RoomType>> = createAction(SET_SELECTED_ROOM);
export const handleJoinRoom: ActionFunctionAny<Action<undefined>> = createAction(HANDLE_JOIN_ROOM);
