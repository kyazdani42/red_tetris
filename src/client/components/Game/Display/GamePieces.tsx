import * as React from 'react';

import { Block } from './Block';
import { GamePiecesContainer } from './styles';

interface Props {
  stack: GameProps['stack'];
}

export const getDivsFromStack = (stack: GameProps['stack']) => {
  const container = Array(200);
  for (let i = 0; i < 20; i++) {
    for (let j = 0; j < 10; j++) {
      const containerIndex = +`${i}${j}`;
      container[containerIndex] = (
        Block(stack[i][j].color, containerIndex + 'indexKey')
      );
    }
  }
  return container;
};

export const GamePieces: React.SFC<Props> = ({ stack }) => (
  <GamePiecesContainer>{getDivsFromStack(stack)}</GamePiecesContainer>
);
