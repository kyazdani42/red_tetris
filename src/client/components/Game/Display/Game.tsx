import * as React from 'react';
import { connect } from 'react-redux';

import { State } from '../../../redux/store';
import { GamePieces } from './GamePieces';
import { NextPiece } from './NextPiece';
import Options from './Options';
import { Result } from './Result';
import { Score } from './Score';
import StartButton from './StartButton';
import { GameStartInfoWrapper } from './styles';

interface Props {
  gameData: GameProps | null;
  socket: SocketIOClient.Socket;
}

export const Game: React.SFC<Props> = ({ gameData, socket }) => {
  let gamePieces, score, nextPiece, startButton, result, options;
  const isPlaying = gameData && gameData.isPlaying || false;
  if (gameData) {
    score = gameData.score;
    nextPiece = gameData.nextPiece;

    const { stack, running, isOwner, winner } = gameData;
    gamePieces = stack.length ? <GamePieces stack={stack} /> : null;
    startButton = running || !isOwner ? null : <StartButton socket={socket} />;
    options = running || !isOwner ? null : <Options />;
    result = winner === undefined || isPlaying ? null : <Result winner={winner} />;
  } else {
    score = 0;
    nextPiece = null;
    gamePieces = null;
    startButton = null;
    result = null;
    options = null;
  }
  return (
    <React.Fragment>
      {gamePieces}
      <Score score={score} />
      <NextPiece nextPiece={nextPiece} />
      {!isPlaying &&
      <GameStartInfoWrapper>
        {startButton}
        {options}
        {result}
      </GameStartInfoWrapper>
      }
    </React.Fragment>
  );
};

const mapStateToProps = ({ app }: State) => ({
  gameData: app.gameData,
  socket: app.socket
});

export default connect(mapStateToProps)(Game);
