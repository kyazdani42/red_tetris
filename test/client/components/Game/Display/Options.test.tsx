// tslint:disable
import * as React from 'react';
import { shallow } from 'enzyme';

import { mapStateToProps, mapDispatchToProps, Options, checkbox } from '../../../../../src/client/components/Game/Display/Options';
import { setOptions } from '../../../../../src/client/redux/actions';

const props = {
  options: {
    invisible: false,
    mirror: false,
    reverse: false,
    speed: false
  },
  dispatchSetOptions: jest.fn()
};

describe('testing the options component', () => {
  it('smoke tests the component', () => {
    const wrapper = shallow(<Options {...props} />)
    expect(wrapper.update()).toMatchSnapshot();
  }) 
})

describe('testing checkbox component', () => {
  const wrapper = shallow(checkbox(true, 'test', () => null, 'displayName'));
  it('test the function', () => {
    expect(wrapper.update()).toMatchSnapshot();
  })
})

describe('testing mapStateToProps', () => {
  const state: any = {
    app: {
      options: 'options'
    }
  }
  it('test the mapping', () => {
    const expected = { options: 'options' };
    expect(mapStateToProps(state)).toEqual(expected);
  })
})

describe('testing mapDispatchToProps', () => {
  it('checks that dispatch is called', () => {
    const dispatch = jest.fn();
    const options = {
      invisible: false,
      mirror: false,
      reverse: false,
      speed: false
    };
    mapDispatchToProps(dispatch).dispatchSetOptions(options);
    expect(dispatch).toHaveBeenCalledWith(setOptions(options));
  })
})

