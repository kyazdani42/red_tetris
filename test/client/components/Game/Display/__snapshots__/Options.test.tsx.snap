// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing checkbox component test the function 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <UNDEFINED>
    <label>
      displayName
    </label>
    <input
      checked={true}
      onChange={[Function]}
      type="checkbox"
    />
  </UNDEFINED>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "children": Array [
        <label>
          displayName
        </label>,
        <input
          checked={true}
          onChange={[Function]}
          type="checkbox"
        />,
      ],
      "forwardedComponent": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-EHOje",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: #fff;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-EHOje",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
      "forwardedRef": null,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "displayName",
        },
        "ref": null,
        "rendered": "displayName",
        "type": "label",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "checked": true,
          "onChange": [Function],
          "type": "checkbox",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "children": Array [
          <label>
            displayName
          </label>,
          <input
            checked={true}
            onChange={[Function]}
            type="checkbox"
          />,
        ],
        "forwardedComponent": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: #fff;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
        "forwardedRef": null,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "displayName",
          },
          "ref": null,
          "rendered": "displayName",
          "type": "label",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "checked": true,
            "onChange": [Function],
            "type": "checkbox",
          },
          "ref": null,
          "rendered": null,
          "type": "input",
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`testing the options component smoke tests the component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Unknown
    dispatchSetOptions={[Function]}
    options={
      Object {
        "invisible": false,
        "mirror": false,
        "reverse": false,
        "speed": false,
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <UNDEFINED>
          <label>
            invisible
          </label>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
        </UNDEFINED>,
        <UNDEFINED>
          <label>
            mirror
          </label>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
        </UNDEFINED>,
        <UNDEFINED>
          <label>
            reverse
          </label>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
        </UNDEFINED>,
        <UNDEFINED>
          <label>
            disable speed
          </label>
          <input
            checked={false}
            onChange={[Function]}
            type="checkbox"
          />
        </UNDEFINED>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <label>
              invisible
            </label>,
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "invisible",
            },
            "ref": null,
            "rendered": "invisible",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "onChange": [Function],
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: #fff;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <label>
              mirror
            </label>,
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "mirror",
            },
            "ref": null,
            "rendered": "mirror",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "onChange": [Function],
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: #fff;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <label>
              reverse
            </label>,
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "reverse",
            },
            "ref": null,
            "rendered": "reverse",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "onChange": [Function],
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: #fff;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "children": Array [
            <label>
              disable speed
            </label>,
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "disable speed",
            },
            "ref": null,
            "rendered": "disable speed",
            "type": "label",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "checked": false,
              "onChange": [Function],
              "type": "checkbox",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-EHOje",
            "isStatic": true,
            "rules": Array [
              "
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: #fff;
",
            ],
          },
          "displayName": "styled.div",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-EHOje",
          "target": "div",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        },
      },
    ],
    "type": Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-ifAKCX",
        "isStatic": true,
        "rules": Array [
          "
  display: flex;
  background-color: rgba(0,0,0,0.5);
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  height: 130px;
  width: 140px;
  position: absolute;
  top: 63%;
  left: 50%;
  margin-left: -70px;
  margin-top: -50px;
",
        ],
      },
      "displayName": "styled.div",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-ifAKCX",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    },
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <UNDEFINED>
            <label>
              invisible
            </label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
          </UNDEFINED>,
          <UNDEFINED>
            <label>
              mirror
            </label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
          </UNDEFINED>,
          <UNDEFINED>
            <label>
              reverse
            </label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
          </UNDEFINED>,
          <UNDEFINED>
            <label>
              disable speed
            </label>
            <input
              checked={false}
              onChange={[Function]}
              type="checkbox"
            />
          </UNDEFINED>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <label>
                invisible
              </label>,
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "invisible",
              },
              "ref": null,
              "rendered": "invisible",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: #fff;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <label>
                mirror
              </label>,
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "mirror",
              },
              "ref": null,
              "rendered": "mirror",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: #fff;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <label>
                reverse
              </label>,
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "reverse",
              },
              "ref": null,
              "rendered": "reverse",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: #fff;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "children": Array [
              <label>
                disable speed
              </label>,
              <input
                checked={false}
                onChange={[Function]}
                type="checkbox"
              />,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "disable speed",
              },
              "ref": null,
              "rendered": "disable speed",
              "type": "label",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "checked": false,
                "onChange": [Function],
                "type": "checkbox",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": Object {
            "$$typeof": Symbol(react.forward_ref),
            "attrs": Array [],
            "componentStyle": ComponentStyle {
              "componentId": "sc-EHOje",
              "isStatic": true,
              "rules": Array [
                "
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: center;
  color: #fff;
",
              ],
            },
            "displayName": "styled.div",
            "foldedComponentIds": Array [],
            "render": [Function],
            "styledComponentId": "sc-EHOje",
            "target": "div",
            "toString": [Function],
            "warnTooManyClasses": [Function],
            "withComponent": [Function],
          },
        },
      ],
      "type": Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-ifAKCX",
          "isStatic": true,
          "rules": Array [
            "
  display: flex;
  background-color: rgba(0,0,0,0.5);
  flex-direction: column;
  justify-content: space-around;
  align-items: center;
  height: 130px;
  width: 140px;
  position: absolute;
  top: 63%;
  left: 50%;
  margin-left: -70px;
  margin-top: -50px;
",
          ],
        },
        "displayName": "styled.div",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-ifAKCX",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      },
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
